arangod/RestHandler/RestCursorHandler.cpp

```diff
@@ -618,10 +623,24 @@ RestStatus RestCursorHandler::generateCursorResult(rest::ResponseCode code) {
     builder.add(StaticStrings::Error, VPackValue(false));
     builder.add(StaticStrings::Code, VPackValue(static_cast<int>(code)));
     builder.close();
+    _cursor->setLastQueryBatchObject(builder);
```

Comment:
We should not store anything if the cursor is not retriable. Otherwise we will only add overhead for something the user didn't ask for.

Instead of copying the Builder object here and storing it in the cursor, we should store the Builder's Buffer in the cursor. We can even create a Buffer on heap using `make_shared`, and then make the Builder point to it while building the result. And afterwards simply store the shared_ptr in the cursor.